x-pack/plugins/snapshot_restore/public/app/services/utils/clean_settings.ts

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { Repository } from '../../../../common/types';
+
+/**
+ * Utility to remove empty fields ("") from repository settings
+ */
+export const cleanSettings = (settings: Repository['settings']) =>
```

Comment:
We might want to be able to use the same method for other forms object? What about naming it something like `removeEmptyStringsFromObject(...)`. It could even do it deeply (with an optional parameter), but then you'd be passing the interview a second time 😄

Reply:
Just to play devil's advocate, I think there is a benefit to preserving this as specialized logic for cleaning the settings object, as opposed to generalizing it. In its current form it's really clear to me what its purpose is: empty strings have a special meaning in the context of the `settings` object, and this logic tells me that we don't want to keep them (for some reason). It's really simple to grok the intention behind this code. I understand that there's a business rule here and if we make changes to this code it will because we're adapting to changes in the business rule.

On the other hand, if we extract this out into a generalized utility, now the context has changed and we've invited new features (like the ability to clean an object deeply, which may not be necessary or even desired for our use case). If our business rules change to where we're using `null` instead of empty strings, then we'd need to either modify the generalized function or write new code.

Based on these pros and cons I really don't think we should bother generalizing this. :)

Reply:
The more I keep reading about functional programming the more we will have a discrepancy on this and the way we build software. 😊 In lean towards functions that **do not** have any business "specialized logic" baked in them. I believe in small functions that do 1 thing well and that are composable (especially utilities functions).

In the **composition** of those small function resides the business logic.

In the long run, it makes code cleaner, the webpage faster (no need to send the same JS over and over in differents .bundle.js), lower memory footprint, and I would even say more robust as the same piece of code is being executed hundreds of time (and thus fully tested). 
And it goes without saying the fact that it saves in development time by not having to re-think, re-write and re-test the same exact code.

I mention it because.... I wrote a very similar function in CCR I think 😊 and the "_empty strings have a special meaning in the context of the settings object_" statement is not totally true. We need it in all those scenarios where we don't want the client to send form fields that haven't been modified.